source/MaterialXGenShader/DefaultColorManagementSystem.cpp

```diff
@@ -0,0 +1,28 @@
+#include <MaterialXGenShader/DefaultColorManagementSystem.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+
+namespace MaterialX
+{
+
+const string DefaultColorManagementSystem::CMS_NAME = "default_cms";
+
+
+string DefaultColorManagementSystem::getImplementationName(const ColorSpaceTransform& transform)
+{
+    string language = _shadergen.getLanguage();
+    std::replace(language.begin(), language.end(), '-', '_');
+    return "IM_" + transform.sourceSpace + "_to_" + transform.targetSpace + "_" + transform.type->getName() + "_" + language;
+}
+
+DefaultColorManagementSystemPtr DefaultColorManagementSystem::create(ShaderGenerator& shadergen)
+{
+    DefaultColorManagementSystemPtr result(new DefaultColorManagementSystem(shadergen));
+    return result;
```

Comment:
Nitpick: for consistency with other creator methods you could use: `return std::make_shared<DefaultColorManagementSystem>(shadergen);`

Reply:
I would love to, but you can't with private constructors: https://stackoverflow.com/questions/8147027/how-do-i-call-stdmake-shared-on-a-class-with-only-protected-or-private-const

Reply:
Ah I didn't realise the constructor was private. All good then :)